File: AtCoder/B_Glass_and_Mug.go
package main

import (
	"fmt"
)

func main() {
	var K, G, M int
	fmt.Scan(&K, &G, &M)

	a, b := G, M
	for K > 0 {
		if a == G {
			a = 0
		} else if b == 0 {
			b = M
		} else {
			x := min(G - a, b)
			a += x
			b -= x
		}
		K--
	}

	fmt.Println(a, b)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: AtCoder/B_Glass_and_Mug.rs
use proconio::input;

fn main() {
    input! {
        k: i32,
        g: i32,
        m: i32,
    }

    let mut a = g;
    let mut b = m;
    for _ in 0..k {
        if a == g {
            a = 0;
        } else if b == 0 {
            b = m;
        } else {
            let x = b.min(g - a);
            a += x;
            b -= x;
        }
    }

    println!("{} {}", a, b);
}